extensions/todoist/src/components/FilterTasks.tsx

```diff
@@ -0,0 +1,69 @@
+import { List, ActionPanel, Action, Icon } from "@raycast/api";
+import { useEffect, useState } from "react";
+
+import { Task, getFilterTasks } from "../../src/api";
+import { filterSort } from "../helpers/filters";
+import { QuickLinkView, ViewMode } from "../home";
+import useCachedData from "../hooks/useCachedData";
+import useViewTasks from "../hooks/useViewTasks";
+
+import CreateViewAction from "./CreateViewAction";
+import TaskListSections from "./TaskListSections";
+
+type FilterTasksProps = { name: string; quickLinkView?: QuickLinkView };
+
+function FilterTasks({ name, quickLinkView }: FilterTasksProps) {
+  const [data] = useCachedData();
+  const filters = data?.filters;
+  const filter = filters?.find((filter) => filter.name === name);
+  const query = filter?.query || "";
+  const [tasks, setTasks] = useState<Task[]>([]);
+
+  useEffect(() => {
+    if (query) {
+      getFilterTasks(query)
+        .then((fetchedTasks) => {
+          setTasks(filterSort(fetchedTasks));
+        })
+        .catch(() => {
+          throw new Error("Error fetching filter tasks");
+        });
+    } else {
+      setTasks([]);
+    }
+  }, [query]);
```

Comment:
Why not using `useCachedPromise`? It'll make the code a bit more readable and you can benefit from caching.

Reply:
thanks for the tip. changed the implementation to make use of useCachedPromise.